knative-operator/pkg/common/util.go

```diff
@@ -103,3 +115,37 @@ func EnsureContainerMemoryLimit(s *operatorv1alpha1.CommonSpec, containerName st
 	})
 	return
 }
+
+// common function to enqueue reconcile requests for resources
+func EnqueueRequestByOwnerAnnotations(ownerNameAnnotationKey, ownerNamespaceAnnotationKey string) handler.EventHandler {
+	enqueueRequests := func() handler.ToRequestsFunc {
+		return func(obj handler.MapObject) []reconcile.Request {
```

Comment:
The intermediary function seems unnecessary.

Just

```golang
enqueueRequests := func(obj handler.MapObject) []reconcile.Request { ...}
return &handler.EnqueueRequestsFromMapFunc{ToRequests: enqueueRequests}
```

Should work.

Reply:
Couldn't make that work. Some golang magic is required.

Let's merge this one and can you send a small PR for that?

Reply:
ack!